code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -1373,16 +1401,94 @@
 
 /obj/item/weapon/reagent_containers/food/drinks/thermos
 	name = "\improper Thermos"
-	desc = "A metal flask which insulates its contents from temperature - keeping hot beverages hot, and cold ones cold."
+	desc = "A metal flask which insulates its contents from temperature - keeping hot beverages hot, and cold ones cold. You can remove its cap to use as a cup."
 	icon_state = "vacuumflask"
 	origin_tech = Tc_MATERIALS + "=1"
 	amount_per_transfer_from_this = 10
 	volume = 100
+	var/obj/item/weapon/reagent_containers/food/drinks/thermos_cap/cap
+
+/obj/item/weapon/reagent_containers/food/drinks/thermos/New()
+	..()
+	cap = new(src)
+
+/obj/item/weapon/reagent_containers/food/drinks/thermos/attack_self(var/mob/user)
+	if (cap)
+		to_chat(user, "<span class='warning'>Remove the cap with your other hand first.</span>")
```

Comment:
perhaps more intuitive to pop the cap off here instead? that way even if your other hand is full you can remove the cap and let it fall to the floor, which is totally doable in any number of ways

Reply:
This is actually a design choice. I don't want players to let the cap fall onto the floor.